packages/black-forest-labs/src/bfl-image-model.ts

```diff
@@ -0,0 +1,280 @@
+  readonly specificationVersion = 'v2';
+  readonly maxImagesPerCall = 1;
+
+  get provider(): string {
+    return this.config.provider;
+  }
+
+  constructor(
+    readonly modelId: BlackForestLabsImageModelId,
+    private readonly config: BlackForestLabsImageModelConfig,
+  ) {}
+
+  async doGenerate({
+    prompt,
+    size,
+    aspectRatio,
+    providerOptions,
+    headers,
+    abortSignal,
+  }: Parameters<ImageModelV2['doGenerate']>[0]): Promise<
+    Awaited<ReturnType<ImageModelV2['doGenerate']>>
+  > {
+    const warnings: Array<ImageModelV2CallWarning> = [];
+
+    const finalAspectRatio =
+      aspectRatio ?? (size ? convertSizeToAspectRatio(size) : undefined);
+
+    if (size && !aspectRatio) {
+      warnings.push({
+        type: 'unsupported-setting',
+        setting: 'size',
+        details:
+          'Black Forest Labs does not accept width/height. Deriving aspect_ratio from size.',
+      });
+    } else if (size && aspectRatio) {
+      warnings.push({
+        type: 'unsupported-setting',
+        setting: 'size',
+        details:
+          'Black Forest Labs ignores size when aspectRatio is provided.',
+      });
+    }
+
+    const currentDate = this.config._internal?.currentDate?.() ?? new Date();
+
+    const submit = await postJsonToApi({
+      url: `${this.config.baseURL}/${this.modelId}`,
+      headers: combineHeaders(await resolve(this.config.headers), headers),
+      body: {
+        prompt,
+        ...(finalAspectRatio ? { aspect_ratio: finalAspectRatio } : {}),
+        ...(providerOptions.bfl ?? {}),
+        ...(providerOptions['black-forest-labs'] ?? {}),
+      },
+      failedResponseHandler: bflFailedResponseHandler,
+      successfulResponseHandler: createJsonResponseHandler(bflSubmitSchema),
+      abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    const pollUrl = submit.value.polling_url;
+    const requestId = submit.value.id;
+    const imageUrl = await this.pollForImageUrl({
+      pollUrl,
+      requestId,
+      abortSignal,
+    });
+
+    const { value: imageBytes, responseHeaders } = await getFromApi({
+      url: imageUrl,
+      abortSignal,
+      failedResponseHandler: createStatusCodeErrorResponseHandler(),
+      successfulResponseHandler: createBinaryResponseHandler(),
+      fetch: this.config.fetch,
+    });
+
+    return {
+      images: [imageBytes],
+      warnings,
+      response: {
+        modelId: this.modelId,
+        timestamp: currentDate,
+        headers: responseHeaders,
+      },
+    };
+  }
+
+  private async pollForImageUrl({
+    pollUrl,
+    requestId,
+    abortSignal,
+  }: {
+    pollUrl: string;
+    requestId: string;
+    abortSignal: AbortSignal | undefined;
+  }): Promise<string> {
+    const url = new URL(pollUrl);
+    if (!url.searchParams.has('id')) {
+      url.searchParams.set('id', requestId);
+    }
+
+    while (true) {
+      const response = await (this.config.fetch ?? fetch)(url.toString(), {
```

Comment:
Can we use `getFromApi` here which has json response handlers? see `get-from-api.ts`